src/solver/clause.rs

```diff
@@ -503,40 +481,33 @@ impl ClauseState {
 /// Represents a literal in a SAT clause (i.e. either A or ¬A)
 #[repr(transparent)]
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
-pub(crate) struct Literal(u32);
+pub(crate) struct Literal(NonZeroU32);
 
 impl Literal {
     /// Constructs a new [`Literal`] from a [`VariableId`] and a boolean
     /// indicating whether the literal should be negated.
     pub fn new(variable: VariableId, negate: bool) -> Self {
         let variable_idx = variable.to_usize();
-        let literal_idx = variable_idx << 1 | negate as usize;
-        Self(literal_idx.try_into().expect("literal id too big"))
+        let encoded_literal = variable_idx << 1 | negate as usize;
+        Self::from_usize(encoded_literal)
     }
 }
 
 impl ArenaId for Literal {
     fn from_usize(x: usize) -> Self {
-        debug_assert!(x <= u32::MAX as usize, "watched literal id too big");
-        Literal(x as u32)
+        let idx: u32 = (x + 1).try_into().expect("watched literal id too big");
+        // SAFETY: This is safe because we are adding 1 to the index
+        unsafe { Literal(NonZeroU32::new_unchecked(idx)) }
     }
 
     fn to_usize(self) -> usize {
```

Comment:
There was a previous `#[inline]` in this PR, if we need it there we need it here I think?

Reply:
I removed all `#[inline]` statements.